src/types.rs

```diff
@@ -347,12 +349,63 @@ pub enum PlayerClass {
     Void,
 }
 
+#[derive(Debug, Clone)]
+pub struct Position {
+    x: f32,
+    y: f32,
+    z: f32,
+}
+
+impl<'de> Deserialize<'de> for Position {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        struct PositonVisitor {}
+
+        impl<'de> Visitor<'de> for PositonVisitor {
+            type Value = Position;
+
+            fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+                formatter.write_str("a 3 element list of f32 seperated by space")
+            }
+
+            fn visit_str<E>(self, s: &str) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                let mut coords = [0f32; 3];
+                for (i, s) in s.split(" ").enumerate() {
+                    coords[i] = s.parse().map_err(|_| serde::de::Error::invalid_type(serde::de::Unexpected::Str(s), &self))?;
+                }
+                let [x, y, z] = coords;
+                Ok(Position { x, y, z })
+            }
+
+            fn visit_string<E>(self, s: String) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                self.visit_str(&s)
+            }
+        }
+
+        deserializer.deserialize_str(PositonVisitor {})
+    }
+}
+
 #[cfg(test)]
 mod tests {
+
     use super::*;
 
     #[test]
     fn test_data_parsable() -> io::Result<()> {
         GameStats::from_dir("test_data").map(|_| ())
     }
+
+    #[test]
+    fn position_deserialize() -> serde_json::Result<()> {
+        serde_json::from_str::<Vec<Position>>("[\"1.0 -1.0 0.1\"]").map(|_| ())
```

Comment:
```suggestion
        serde_json::from_str::<Position>("\"1.0 -1.0 0.1\"").map(|_| ())
```

Reply:
afaik this doesn't work, since `serde_json` is expecting a JSON object there.

Reply:
It seems to work for me.

Reply:
Then I made a mistake before. I will change that